Eastern-Research-Group/csb-rebate-forms-app
# Feature/update user data check

## Related Issues:
* CSBAPP-62

## Main Changes:
Replaces final piece of fetched data management via a React Context Component (fetching user data – also used for authentication) with react-query managed fetched data.

## Steps To Test:
1. Navigate to a form you can edit (preferably open up an "Edit Grid" component) and open the browser's network tab.
2. Wait 12 minutes, then move the mouse – you should see a request happen to `/api/user` in the background to refresh the JWT, and the "Edit Grid" component should remain open.
3. Now wait 14 minutes, and an "Inactivity Warning" dialog should appear. Click the "Stay logged in" button and you should see a request happen to `/api/user` to refresh the JWT, and the "Edit Grid" component should remain open.
4. Repeat the above step, but allow the countdown timer to reach zero. You should be logged out of the app and re-directed to the homepage with "For security reasons, you have been logged out due to 15 minutes of inactivity." displayed.
